test/suite/wttest.py

```diff
@@ -226,15 +234,30 @@ def globalSetup(preserveFiles = False, removeAtStart = True, useTimestamp = Fals
         WiredTigerTestCase._stdout = sys.stdout
         WiredTigerTestCase._stderr = sys.stderr
         WiredTigerTestCase._concurrent = False
-        WiredTigerTestCase._globalSetup = True
         WiredTigerTestCase._seeds = [521288629, 362436069]
         WiredTigerTestCase._randomseed = False
+        WiredTigerTestCase._retriesAfterRollback = 0
+        WiredTigerTestCase._testsRun = 0
         if hookmgr == None:
             hookmgr = wthooks.WiredTigerHookManager()
         WiredTigerTestCase._hookmgr = hookmgr
         if seedw != 0 and seedz != 0:
             WiredTigerTestCase._randomseed = True
             WiredTigerTestCase._seeds = [seedw, seedz]
+        WiredTigerTestCase._globalSetup = True
+
+    @staticmethod
+    def finalReport():
+        # We retry tests that get rollback errors in a way that is mostly invisible.
+        # self._rollbacksAllowedPerTest makes sure a single test doesn't rollback too many times.
+        # We also want to report an error if the total number of retries gets above a threshold,
+        # we use 1% if the number of tests is large enough.  Each rollback error's stack
+        # is already recorded, so we give visibility to it here.
+        total = WiredTigerTestCase._testsRun
+        retries = WiredTigerTestCase._retriesAfterRollback
+        if (total > 300 and retries/total > 0.01) or (total <= 300 and retries >= 3):
```

Comment:
It is unclear to me what 300 means here and how/why that number was chosen? Please clarify in the comment and/or make it a macro or define of some kind. To me it could mean any of:
- total number of tests
- number of tests run so far
- number of retries
- seconds

Reply:
I've rewritten this code and added a comment.  Hopefully will be much clearer - please check on the re-review.